lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -54,13 +57,33 @@ def timeout_message
     end
 
     def deploy_timed_out?
-      super || @latest_rs && @latest_rs.deploy_timed_out?
+      if @progress
+        @progress["status"] == 'False'
+      else
+        super || @latest_rs && @latest_rs.deploy_timed_out?
```

Comment:
I am not entirely sure what you mean here? The current logic falls back to the default logic if the progress condition does not exist. Isn't that what we want?